### PR TITLE
Update ChainerMN documents

### LINKED ISSUE #5297: Update ChainerMN documents
- [x] Update [ChainerMN installation guide](https://docs.chainer.org/en/latest/chainermn/installation/guide.html)
- [x] Add a brief summary of ChainerMN in https://docs.chainer.org/en/latest/chainermn/index.html; how about adding a content of https://github.com/chainer/chainermn/blob/master/README.md?
- [x] Update Upgrade Guide to encourage users to install `chainermn` package before upgrading to v5.

Ref #5226.

### PR DESCRIPTION
**Pull request by kmaehashi**
Fix #5297

Merge #5300 first.

### CODE REVIEW

**Comment by kmaehashi** [comment]
Rebased to the latest master.

Jenkins, test this please.

File: docs/source/tmp.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import matplotlib
```
Comment (keisukefukuda) [inline docs/source/tmp.py]: I'm not sure files under `docs/source` are ChainerMN-related...  Are they?
?_?

**Comment by kmaehashi** [comment]
Test failures seems not related to this PR.

**Comment by kuenishi** [comment]
👍 to merge.

**Pull request merged**